server/test/medium/specs/user.service.spec.ts

```diff
@@ -0,0 +1,65 @@
+import { UserService } from 'src/services/user.service';
+import { TestContext, TestFactory } from 'test/factory';
+import { getKyselyDB, newTestService } from 'test/utils';
+
+describe(UserService.name, () => {
+  let sut: UserService;
+  let context: TestContext;
+
+  beforeAll(async () => {
+    const db = await getKyselyDB();
+    context = await TestContext.from(db).withUser({ isAdmin: true }).create();
+    ({ sut } = newTestService(UserService, context));
+  });
+
+  describe('create', () => {
+    it('should create a user', async () => {
+      const userDto = TestFactory.user();
+
+      await expect(sut.createUser(userDto)).resolves.toEqual(
+        expect.objectContaining({
+          id: userDto.id,
+          name: userDto.name,
+          email: userDto.email,
+        }),
+      );
+    });
+  });
+
+  describe('get', () => {
+    it('should get a user', async () => {
+      const userDto = TestFactory.user();
+
+      await context.createUser(userDto);
+
+      await expect(sut.get(userDto.id)).resolves.toEqual(
+        expect.objectContaining({
+          id: userDto.id,
+          name: userDto.name,
+          email: userDto.email,
+        }),
+      );
+    });
+  });
+
+  describe('updateMe', () => {
+    it('should update a user', async () => {
+      const userDto = TestFactory.user();
+      const sessionDto = TestFactory.session({ userId: userDto.id });
+      const authDto = TestFactory.auth({ user: userDto });
```

Comment:
This seems to be a popular combination. Maybe it's worth it adding some utility function for that that returns you a new user, session, and auth, all already linked?

Reply:
maybe later, we'll see how often it comes up